### ISSUE TITLE
How is val_loss calculated?

### ISSUE THREAD

**Issue by AleffSatz** [matterport/Mask_RCNN#2675]
Hi, just wondering how the val_loss is calculated. My assumption is val_loss is the sum of the other losses rpn_class_losses, rpn_box_losses etc but this doesn't seem the case in the train_shapes.ipynb. 

In particular, the train_loss, val_loss for the train-all-layer phase seems to skyrocket while the other losses remain roughly stable compared to the train-head-only phase

Train head: val loss is 1.854
![image](https://user-images.githubusercontent.com/34236643/130341654-f8a87863-a5a8-4071-ab68-27b91cc646a6.png)

Train all layers: val_loss is 11.4 while the other losses actually decrease
![image](https://user-images.githubusercontent.com/34236643/130341565-5eb2802b-3f8f-4660-a79a-90db138d876a.png)

**Comment by agbozo1** [comment]
Someone placed the answer on StackOverflow:
[https://stackoverflow.com/a/51218888/9161828](https://stackoverflow.com/a/51218888/9161828
)
Hope it helps you

**Comment by Behnam72** [comment]
> https://stackoverflow.com/a/51218888/9161828

I have a similar issue. Were you able to solve it?

**Comment by Behnam72** [comment]
I have a similar issue, could you solve it?

**Comment by AleffSatz** [comment]
Not really, never found out the reason. I just ended up computing a custom loss as the sum of those losses if I recall correctly.
